### PR TITLE
[RLlib] more connector polishes and fixes.

### PR DESCRIPTION
**Pull request by gjoliver**
Most importantly:
* Fix preprocessing enabling logic.
* Target network mixin shouldn't override get_variables().
* When serializing obs space, we also need to serialize original_space.
* Introduce a config to force enable TF2 eager exec even if framework is not tf2.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [*] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/examples/policy/random_policy.py [hunk 1]
```diff
@@ -30,6 +30,14 @@ def __init__(self, *args, **kwargs):
         else:
             self.action_space_for_sampling = self.action_space
 
+        # Disable view requirements for all columns except OBS, which is
```
Comment (kouroshHakha) [inline rllib/examples/policy/random_policy.py:33]: **question**: why is this needed? seems a bit ad-hoc to exclude obs and next_obs.

**Review by kouroshHakha — APPROVED** [review]
LGTM as long as it passes the tests. @gjoliver For these multi-feature PRs can you break them down to several individual PRs. I know it's more work but it will be easier to manager / revert later.

File: rllib/examples/policy/random_policy.py (hunk 1 above)
Comment (gjoliver) [reply rllib/examples/policy/random_policy.py:33]: notice obs column is actually needed by this policy below.
this is a custom implemented policy, and doesn't inherit DynamicTF or EagerTF or Torch policies.
so it doesn't have the "automatically figuring out which columns are needed for training and for action computation" functionality.
the VR specs then need to be manually specified, otherwise Connector will try to stack infos and other columns.

**Comment by gjoliver** [comment]
> LGTM as long as it passes the tests. @gjoliver For these multi-feature PRs can you break them down to several individual PRs. I know it's more work but it will be easier to manager / revert later.

for sure, I think that is a really really good habit.
to be honest I just don't know how PR stacking works in Git world, like I do need all these changes for the thing to work e2e, so they need to co-exist in a single branch, but also get sent as separate PRs. something I never learned.

File: rllib/examples/policy/random_policy.py (hunk 1 above)
Comment (kouroshHakha) [reply rllib/examples/policy/random_policy.py:33]: Then why don't you override the `init_viewrequirements()` function and do this there?
```
def init_view_requirements(self):
    super().init_view_requirements()
    # Disable view requirements for all columns except OBS, which is
    # actually used.
    for k, v in self.view_requirements.items():
        if k == SampleBatch.OBS or k == SampleBatch.NEXT_OBS:
            continue
        v.used_for_training = False
        v.used_for_compute_actions = False
```

Comment (gjoliver) [reply rllib/examples/policy/random_policy.py:33]: oh yeah ... this is exactly why we have code reviews :)
thanks
